File: backend/MiniOrder.Domain/Entities/Product.cs
namespace MiniOrder.Domain.Entities;

public class Product
{
    public int Id { get; set; }

    public int CategoryId { get; set; }

    public string StockCode { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public int StockQuantity { get; set; }

    public bool IsActive { get; set; } = true;

    public Category Category { get; set; } = null!;

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/product.ts
export interface Product {
  id: number;
  stockCode: string;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  categoryId: number;
  categoryName: string;
  brandName: string;
  coverImageUrl: string | null;
}

export interface ProductImage {
  imageUrl: string;
  isCover: boolean;
  sortOrder: number;
}

export interface ProductAttribute {
  name: string;
  code: string;
  dataType: string;
  value: string;
  sortOrder: number;
}

export interface ProductDetail {
  id: number;
  stockCode: string;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  categoryId: number;
  categoryName: string;
  parentCategoryName: string | null;
  brandId: number;
  brandName: string;
  images: ProductImage[];
  attributes: ProductAttribute[];
}
